Add icon to layer and fields comboboxes in "map value" widget's Load …

DelazJ opened the pull request:
…field from layer dialog

Before / After

![image](https://github.com/user-attachments/assets/0a4376ea-2be7-481f-91fc-29452e677dc0)

DelazJ committed:
Add icon to layer and fields comboboxes in "map value" widget's Load field from layer dialog

nyalldawson commented:
Hmm... any reason why we wouldn't want to use the standard QgsMapLayerComboBox/ QgsFieldCombo widgets here?

DelazJ commented:
I simply forgot they exist. Thanks for the reminder. I will do the update.

DelazJ committed:
Use dedicated QgsMapLayerComboBox and QgsFieldComboBox classes

to handle layer and fields widgets

DelazJ committed:
Add hep button and remember dialog placement

DelazJ commented:
I moved the widgets to use the dedicated field and map widgets and much better in the way the preview table is populated. And while at it, I added a help button and dialog geometry restoring.
I've played a lot with that dialog and i find the behavior quite confusing, but didn't address it here. This PR keeps the current behavior. See #59386 for the details on the issue.

nyalldawson commented on src/gui/qgsattributetypeloaddialog.cpp:
```diff
@@ -120,9 +82,9 @@ void QgsAttributeTypeLoadDialog::createPreview( int fieldIndex, bool full )
-  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentData().toString() );
+  const int idx = keyComboBox->currentIndex();
+  const int idx2 = valueComboBox->currentIndex();
+  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentLayer()->id() );
```
```suggestion
  QgsMapLayer *dataLayer = layerComboBox->currentLayer();
```

nyalldawson commented on src/gui/qgsattributetypeloaddialog.cpp:
```diff
@@ -169,9 +131,9 @@ bool QgsAttributeTypeLoadDialog::insertNull()
-  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentData().toString() );
+  const int idx = keyComboBox->currentIndex();
+  const int idx2 = valueComboBox->currentIndex();
+  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentLayer()->id() );
```
```suggestion
  QgsMapLayer *dataLayer = layerComboBox->currentLayer();
```

nyalldawson committed:
Apply suggestions from code review
